mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SmartTestingMavenConfigurer.java

```diff
@@ -54,17 +59,38 @@ public void afterProjectsRead(MavenSession session) throws MavenExecutionExcepti
 
         if (configuration.areStrategiesDefined()) {
             configureExtension(session, configuration);
+            storeConfiguration(session);
             calculateChanges();
             Runtime.getRuntime().addShutdownHook(new Thread(() -> purgeLocalStorage(session)));
         } else {
             logStrategiesNotDefined();
         }
     }
 
+    private void storeConfiguration(MavenSession session) {
+        final Path path = configuration.dump();
+        session.getAllProjects().forEach(mavenProject -> copyConfigurationFile(mavenProject.getModel(), path.toFile()));
```

Comment:
We already go through all projects in `configureExtension(session, configuration);` - don't loop twice